packages/hadron-app-registry/src/register-plugin.tsx

```diff
@@ -78,8 +88,29 @@ export type HadronPluginConfig<T, S extends Record<string, () => unknown>> = {
   };
 };
 
-export type HadronPluginComponent<T> = React.FunctionComponent<T> & {
+export type HadronPluginComponent<
+  T,
+  S extends Record<string, () => unknown>
+> = React.FunctionComponent<T> & {
   displayName: string;
+  /**
+   * Convenience method for testing: allows to override services and app
+   * registries available in the plugin context
+   *
+   * @example
+   * const PluginWithLogger = registerHadronPlugin({ ... }, { logger: loggerLocator });
+   *
+   * const MockPlugin = PluginWithLogger.withMockServices({ logger: Sinon.stub() });
+   *
+   * @param mocks Overrides for the services locator values and registries
+   *              passed to the plugin in runtime. When `globalAppRegistry`
+   *              override is passed, it will be also used for the
+   *              localAppRegistry override unless `localAppRegistry` is also
+   *              explicitly passed as a mock service.
+   */
+  withMockServices(
+    mocks: Partial<Registries & Services<S>>
+  ): React.FunctionComponent<T>;
```

Comment:
@addaleax this is this new testing helper I mentioned, I also switched drop-namespace tests to use it too (although due to how this particular plugin works, it doesn't make that much of a difference in that case)